ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -577,26 +675,117 @@ private object OracleQueries extends Queries {
       pvs: Put[Vector[String]],
   ): T[Query0[DBContract[Mark, JsValue, JsValue, Vector[String]]]] = {
     val Seq((tpid, predicate @ _)) = queries // TODO SC handle more than one
-    val _ = parties
     println("selecting")
-    val q = sql"""SELECT contract_id, key, payload, agreement_text
-                  FROM contract
-                  WHERE tpid = $tpid""" // TODO SC AND (""" ++ predicate ++ sql")"
+    import Queries.CompatImplicits.catsReducibleFromFoldable1
+    // % is explicitly reserved by specification as a delimiter
+    val q = sql"""SELECT c.contract_id, key, payload, agreement_text, sd.parties, od.parties
+                  FROM (contract c
+                        JOIN signatories sm ON (c.contract_id = sm.contract_id)
+                        JOIN observers om ON (c.contract_id = om.contract_id))
```

Comment:
what happens if there are no observers or signatories for a given contract? Doesn’t the inner join here just filter them out?

Reply:
I see, yes. I'll address that in the followup.

Reply:
Or 07059d0ac4b197dbc3568bc3e333fa8b2c84cead actually